pkg/controllers/reconciler_test.go

```diff
@@ -61,18 +67,33 @@ func TestUpgrade(test *testing.T) {
 			expectedInitimage: initimage,
 			upgrade:           false,
 		},
+		{
+			envVars: map[string]string{strings.ReplaceAll(strings.ToUpper(filepath.Base(image)), "-", "_") + "_SHA": image + "@sha256:" + sha256,
+				strings.ReplaceAll(strings.ToUpper(filepath.Base(initimage)), "-", "_") + "_SHA": initimage + "@sha256:" + sha256},
+			image:             image + "@sha256:" + preSha256,
+			expectedImage:     image + "@sha256:" + sha256,
+			initimage:         initimage + "@sha256:" + preSha256,
+			expectedInitimage: initimage + "@sha256:" + sha256,
+			upgrade:           true,
+		},
 	}
 
 	for i := range tests {
 		t := tests[i]
 
+		for key, value := range t.envVars {
+			os.Setenv(key, value)
+		}
+
 		upgrade := UpgradeImages(context.Background(), &t.image, &t.initimage)
 
 		if !(upgrade == t.upgrade && t.image == t.expectedImage && t.initimage == t.expectedInitimage) {
 			test.Errorf("expectedUpgrade: %v, received: %v", t.upgrade, upgrade)
 			test.Errorf("expectedImage: %s, received: %s", t.expectedImage, t.image)
 			test.Errorf("expectedInitimage: %s, received: %s", t.expectedInitimage, t.initimage)
 		}
+
+		os.Clearenv()
```

Comment:
This will delete everything, it should only remove the ones that were added.

Reply:
fixed to `os.UnsetEnv`